### PR TITLE
jameica: ensure gsetting schema is available

### LINKED ISSUE #182863: jameica: crash with No GSettings schemas are installed
### Describe the bug
When a user starts [jameica](https://search.nixos.org/packages?channel=22.05&show=jameica&from=0&size=50&sort=relevance&type=packages&query=jameica) and opens a file dialog the application crashes with `(Jameica 2.10.2:1041067): GLib-GIO-ERROR **: 22:23:21.974: No GSettings schemas are installed on the system
`

### Steps To Reproduce
1. Open jameica
2. create a "workspace"
3. file -> manage backups -> restore different backup

### Expected behavior
Opens a gtk file dialog

### Additional context
jameica crashes on every file dialog for me. the backup path is just the easiest accessible and something a user may come along quickly.

```
(Jameica 2.10.2:1055961): Gtk-CRITICAL **: 22:32:14.066: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar
[Mon Jul 25 22:32:13 CEST 2022][INFO][main][de.willuhn.jameica.hbci.synchronize.SynchronizeEngine.getBackends] loading synchronize backends
[Mon Jul 25 22:32:14 CEST 2022][INFO][main][de.willuhn.jameica.hbci.synchronize.AbstractSynchronizeBackend.getJobProviders] loading synchronize providers for backend FinTS/HBCI
[Mon Jul 25 22:32:14 CEST 2022][INFO][main][de.willuhn.jameica.hbci.synchronize.AbstractSynchronizeBackend.getJobProviders]   found 6 provider(s)
[Mon Jul 25 22:32:14 CEST 2022][INFO][main][de.willuhn.jameica.hbci.synchronize.AbstractSynchronizeBackend.getJobProviders] loading synchronize providers for backend Scripting
[Mon Jul 25 22:32:14 CEST 2022][INFO][main][de.willuhn.jameica.hbci.synchronize.AbstractSynchronizeBackend.getJobProviders]   found 1 provider(s)
[Mon Jul 25 22:32:14 CEST 2022][INFO][main][de.willuhn.jameica.hbci.synchronize.AbstractSynchronizeBackend.getJobProviders] loading synchronize providers for backend non-HBCI (Hibiscus Mashup)
[Mon Jul 25 22:32:14 CEST 2022][INFO][main][de.willuhn.jameica.hbci.synchronize.AbstractSynchronizeBackend.getJobProviders]   found 2 provider(s)

(Jameica 2.10.2:1055961): Gtk-CRITICAL **: 22:32:17.057: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(Jameica 2.10.2:1055961): GLib-GIO-ERROR **: 22:32:17.965: No GSettings schemas are installed on the system
[1]    1055961 trace trap (core dumped)  jameica

```
i was already playing around a bit with the wrapGAppsHook, but i just dont get it running atm. 
so maybe someone else want to take a look on that.

i am running a kde session on wayland, but i doubt that it make any difference on gnome or other DE.
 

### Notify maintainers

@flokli @r3dl3g 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
shawn•~» nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.18.11-xanmod1, NixOS, 22.11 (Raccoon), 22.11.20220723.42ca9be`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - channels(shawn): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/v30d42gs5j2x46486d3jrx23hr9qhdvh-source`
```

### PR DESCRIPTION
**Pull request by Shawn8901**
###### Description of changes

this fixes #182863 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by flokli** [comment]
Is there a way to prevent the double wrapping from happening? @SuperSandro2000 ?

**Comment by Shawn8901** [comment]
> Is there a way to prevent the double wrapping from happening? @SuperSandro2000 ?

Your absolute right. i have checked other situations like this, and it should now no longer be double wrapped if i see it correct.

File: pkgs/applications/office/jameica/default.nix [hunk 1]
```diff
@@ -35,13 +35,17 @@ stdenv.mkDerivation rec {
...
+  dontWrapGApps = true;
+
   # there is also a build.gradle, but it only seems to be used to vendor 3rd party libraries
   # and is not able to build the application itself
   buildPhase = ''
     (cd build; ant -Dsystem.version=${version} init compile jar)
   '';
 
   installPhase = ''
+    gappsWrapperArgsHook
```
Comment (SuperSandro2000) [inline pkgs/applications/office/jameica/default.nix:47]: That is strange. Normally we don't need to run this. Are we missing some runHook's? If so please add them instead.

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise right approach

File: pkgs/applications/office/jameica/default.nix (hunk 1 above)
Comment (Shawn8901) [reply pkgs/applications/office/jameica/default.nix:47]: ill check that. 
i did already check if it does make a difference, as the `wrapGAppsHook` is already added.
And yes it makes a difference, when `gappsWrapperArgsHook` is called, it additionally adds the needed `GDK_PIXBUF_MODULE_FILE` and `XDG_DATA_DIRS` which are the variables i am looking for. But atm idk why they are missing when just the hook does its job.

Comment (Shawn8901) [reply pkgs/applications/office/jameica/default.nix]: okay, the `wrapGAppsHook` fills the `gappsWrapperArgsHook` in preFixup, which is the reason, that the args are unset when using them in `installPhase`. 
So i moved the `makeWrapper` call to a phase after that (its mostly done in fixup or postFixup), in my case i opted for fixupPhase.

Hopefully thats okay, if its not, please let me know.

File: pkgs/applications/office/jameica/default.nix [hunk 2]
```diff
@@ -53,13 +55,16 @@ stdenv.mkDerivation rec {
     install -Dm644 plugin.xml $out/share/java/
     install -Dm644 build/jameica-icon.png $out/share/pixmaps/jameica.png
     cp ${desktopItem}/share/applications/* $out/share/applications/
+  '';
 
+  fixupPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/office/jameica/default.nix:60]:
```suggestion
  postFixup = ''
```
we never want to overwrite such complicated and important phases

Comment (Shawn8901) [reply pkgs/applications/office/jameica/default.nix]: am i assuming right, that PRs that change this for other packages are then welcome? as i wrote there are others like that.

Comment (SuperSandro2000) [reply pkgs/applications/office/jameica/default.nix]: Most packages should not overwrite fixupPhase or patchPhase. There are probably some exceptions and we should comment those. Feel free to do some PRs to fix some things up.

**Pull request merged**